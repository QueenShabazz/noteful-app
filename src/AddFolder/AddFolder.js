import React, { Component } from "react";
import Form from "../Form/Form";
import PropTypes from "prop-types";

export default class AddFolder extends Component {
  render() {
    return (
      <section>
        <h2> Create Folder</h2>
        <Form
          onSubmit={e => {
            this.props.folderSubmit(e);
            if (this.props.folder) {
              this.props.history.push("/");
            }
          }}
        >
          <label htmlFor="folder-name-input">Name</label>
          <input
            type="text"
            id="folder-name-input"
            value={this.props.folder}
            onChange={this.props.newFolder}
          />
          {this.props.nameError}
          <button type="submit">Add Folder</button>
        </Form>
      </section>
    );
  }
}

AddFolder.propTypes = {
  folder: PropTypes.string,
  folderSubmit: PropTypes.func,
  newFolder: PropTypes.func
};
