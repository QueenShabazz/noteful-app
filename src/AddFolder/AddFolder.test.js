import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import AddFolder from './AddFolder'
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() });

describe(`AddItemForm component`, () => {
  it('renders the form', () => {
      
    const wrapper = shallow(
        <AddFolder folder="test" />
        )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})