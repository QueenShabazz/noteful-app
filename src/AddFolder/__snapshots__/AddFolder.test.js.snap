// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddItemForm component renders the form 1`] = `
<section>
  <h2>
     Create Folder
  </h2>
  <Form
    onSubmit={[Function]}
  >
    <label
      htmlFor="folder-name-input"
    >
      Name
    </label>
    <input
      id="folder-name-input"
      type="text"
      value="test"
    />
    <button
      type="submit"
    >
      Add Folder
    </button>
  </Form>
</section>
`;
